HW3/task3.js

```diff
@@ -0,0 +1,117 @@
+// 1.  Реализовать такую функцию sum что бы она выполнялась при условии:
+// 1.1   sum() === 0;
+// 1.2   sum(3)() === 3;
+// 1.3   sum(1)(3)(7)() === 11;
+function sum(a = 0) {
+    let toVal = a;
+    return a === 0 ? 0 : add; // ternary not.
+
+    function add(b) {
+        if (b === undefined) return toVal;
+        toVal += b;
+        return add;
+    }
+}
```

Comment:
1) " a" / "b" / "toVal" not the best names, try to give more informative names. in this case you can use "argument"
2) just as a spelling option "return a ?? add"
3) comment "11" is not informative, it is not clear what exactly you wanted to say